src/modules/verifiable-credentials/verifiable-credentials-base.service.ts

```diff
@@ -108,18 +113,21 @@ export abstract class VerifiableCredentialsServiceBase {
    * @description The type of the exchange. Only vc-api exchanges currently supported.
    * @returns credentials query with matching verifiable presentations
    */
-  async initiateExchange({ type, url }: ExchangeInvitation) {
+  async initiateExchange({
+    type,
+    url,
+  }: ExchangeInvitation): Promise<ContinueExchangeSelections> {
     if (type !== VC_API_EXCHANGE) {
       throw new Error('Only VC-API exchange is supported');
     }
 
     const { data: vpRequest } = await axios.post<VpRequest>(url);
-    const presentationDefinitions = vpRequest.query.find(
+    const credentialQuery = vpRequest.query.find(
```

Comment:
I am agree we should verify this. But we can go even further and validate whole vpRequest with class-validator. Do you think it is worth it or on this stage it is enough null checks of query and credentialQuery?

Reply:
Oh I think up to you. I think even if we use class-validator, it would still throw an error here just from doing a .find on undefined or null. Sorry if I'm misunderstanding your question.

Reply:
Yes, it will still throw an error. The benefit is that several null-checks will be replaced with one validation